internal/pkg/agent/cmd/container_test.go

```diff
@@ -56,3 +56,66 @@ func TestContainerTestPaths(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildEnrollArgs(t *testing.T) {
+	cases := []struct {
```

Comment:
Nit: consider using `map[string]struct` where the key of the map is the name of the test.  The benefit here is that the map of test cases will be iterated over in random order, so if there's any dependency between test cases (there shouldn't be) it will be caught more easily.